i18n/en.json

```
@@ -64,6 +64,10 @@
         "one": "Decode {{ .input }} as a protocol buffer (Protobuf).",
         "other": "Decode every element in {{ .mapInput }} as a protocol buffer (Protobuf)."
     },
+    "decoding-step-function-decodeJwt": {
+        "one": "Decode {{ .input }} as a JSON Web Token (JWT).",
+        "other": "Decode every element in {{ .mapInput }} as a JSON Web Token (JWT)."
```

```suggestion
        "one": "Decode {{ .input }} from a JSON Web Token (JWT).",
        "other": "Decode every element in {{ .mapInput }} from a JSON Web Token (JWT)."
```

I am not sure which preposition fits, but ‘as’ feels wrong.

---

But this is what "from" canonically is supposed to mean, see e.g. https://www.urldecoder.org/

---

Setting aside that I wouldn't accept a random website as "canonical" (:P), you decode the _output_ (unencoded message) from [an encoding], not the input (encoded message) as in your suggested phrasing.

Unlike our sentence, your canonical source doesn't have a subject in any sentence where it uses "decode from".

---

But in any case, this discussion is not specific to JWT. We are using "as" in all the other decode function translations as well.

---

I think maybe we should remove the prepostions altogether.

---

Huh, what would that look like? But I genuinely don't see the issue you have with "as".